package.json

```diff
@@ -41,6 +41,7 @@
     "cli-table3": "^0.6.0",
     "eslint-plugin-simple-import-sort": "6",
     "events": "3.3.0",
+    "factory.ts": "^0.5.2",
```

Comment:
Would it be possible not to add a new dependency for testing? I was thinking to have the tests somewhat like they were added here https://github.com/yearn/yearn-sdk/pull/197

Reply:
I"ve added the same dependency in https://github.com/yearn/yearn-watch/pull/193, it's very useful to create objects and avoid repetition. The tests on https://github.com/yearn/yearn-sdk/pull/197 seem a bit different as they seem to be testing that there're errors thrown and for more complex testing it would be hard to always do

```typescript
tokenMock.mockReturnValueOnce(Promise.resolve("0x001"));
```

when you can do

```typescript
tokenFactory.build()
```

and pass whatever values you'd need

```typescript
export const tokenFactory = Factory.Sync.makeFactory<Token>({
  address: "0x001",
  decimals: "18",
  symbol: "DEAD",
  name: "Dead Token",
  priceUsdc: "0",
  supported: {
    zapper: true
  }
});
```

I can refactor and mock everything, but wondering what are the reasons for avoiding helpful dev dependencies

Reply:
I think the drawback with adding dependencies is it's a bit of extra overhead for contributors, it can save code but it can also make things confusing and more difficult to maintain sometimes. And ideally the sdk should be lightweight. 

@FiboApe what do you think about adding this dependency? If you would use it and you think it's useful then I think we should go ahead and add it

Reply:
How would we want these factories to look like? I'm happy to refactor and just do our own factories, how about something like this https://github.com/yearn/yearn-sdk/pull/218/commits/42f6a228d1b391c4029bcb6658ccda3917e4f981

I can refactor the other factories to use that pattern and remove the `factory.ts` dependency if we prefer this approach, or open to suggestions if someone has a better approach in mind.